lib/apiservers/engine/backends/container.go

```diff
@@ -1070,13 +1092,47 @@ func (c *Container) Containers(config *types.ContainerListOptions) ([]*types.Con
 		}
 
 		// get the docker friendly status
-		_, status := dockerStatus(
+		exitCode, status := dockerStatus(
 			int(t.ProcessConfig.ExitCode),
 			t.ProcessConfig.Status,
 			t.ContainerConfig.State,
 			started,
 			stopped)
 
+		// labels func requires a config be passed
+		// TODO: refactor labelsFromAnnotations func for broader use
+		tempConfig := &containertypes.Config{}
+		err = labelsFromAnnotations(tempConfig, t.ContainerConfig.Annotations)
+		if err != nil && config.Filter.Include("label") {
+			return nil, fmt.Errorf("unable to convert vic annotations to docker labels (%s)", t.ContainerConfig.ContainerID)
+		}
+
+		listContext.Labels = tempConfig.Labels
+		listContext.ExitCode = exitCode
+		listContext.ID = t.ContainerConfig.ContainerID
+
+		// prior to further conversion lets determine if this container
+		// is needed or if the list is complete -- if the container is
+		// needed conversion will continue and the container will be added to the
+		// return array
+		action := filter.IncludeContainer(listContext, t)
+		switch action {
+		case filter.ExcludeAction:
+			// skip to next container
+			continue payloadLoop
+		case filter.StopAction:
+			// we're done
+			break payloadLoop
```

Comment:
The break w/o label will only break out of the switch and not the loop, thus the label